### DISCUSSION TITLE
Rebase Merge と Squash Mergeどっちがいいんだろう

### DISCUSSION THREAD

**Discussion by ryo-morimoto** [ryo-morimoto/liveevent-taggerhelper#3] [General]
いまはRebase Mergeでやってます。

こっちはこういうメリットがあるよ！とか、こっちのほうがこういう面でいいからこっちにすべきじゃない？
とかとかあればお願いします。

**Comment by staticWagomU** [comment]
自分は個人開発ではSquash mergeを使っています。
Squash mergeの利点としては、

- コミットログがきれいになる
- prが1つのコミットにまとまるので、トピックブランチ内のコミットを気にしなくていい（prのタイトルと内容は気をつける）
という利点があります。
しかし、デメリットとしては、

- prが1つのコミットにまとまるので巨大なコミットが出来上がる
- 上記問題に伴って発生する、blameで追いにくい
- ベースブランチの更新を追うときにコンフリクトが起きやすい

という問題があります。
自分としては上２つのメリットに魅力を感じている点、個人開発であればデメリットの「コンフリクトが起きやすい」という現象があまり発生しない点をふまえて、Squash mergeを使っています。

**Reply by ryo-morimoto** [reply]
Squashのメリットわかりやすいですね！
rebaseだと、commit一つ一つを切り分けても動くようにしておかないとメリットが少ないと漠然と考えており。
それこそ、そのレベルのコミット分割を複数人でやることは難しいと思うので、場に出ている情報での今の結論はSquash Mergeですかね～

対局の話をしているところもあるみたいですね。
Suquash Merge派
[squash and mergeしか使ってないけど全く困ってない](https://wp.jmuk.org/2023/11/30/squash-and-merge%E3%81%97%E3%81%8B%E4%BD%BF%E3%81%A3%E3%81%A6%E3%81%AA%E3%81%84%E3%81%91%E3%81%A9%E5%85%A8%E3%81%8F%E5%9B%B0%E3%81%A3%E3%81%A6%E3%81%AA%E3%81%84/)
[Git の Merge戦略・方法について (Squash Mergeなど)](https://zenn.dev/manase/scraps/09ee3f20342eba)

Merge Commit派
https://tech.mobilefactory.jp/entry/2023/11/29/160000
https://www.wantedly.com/companies/socialdog/post_articles/461640

Rebase Merge派閥はいなかった。。。

**Reply by ryo-morimoto** [reply]
Rebaseちょっといけるんちゃう？って方はいてました。
https://fedibird.com/@matsuu/111514853951572681

**Reply by staticWagomU** [reply]
main<-develop<-featureという方向でマージする場合においては、develop<-featureはSquash mergeでmain<-developはrebase and mergeっていうやり方は結構ありです！
ただ、全部のブランチをrebase and mergeするならMacnの一本運用とあまり変わらないのでは？と思います。

**Reply by staticWagomU** [reply]
ねもはもないですが、結局はそれぞれにメリットデメリットあるので、環境に合わせるのが一番ですね

**Reply by ryo-morimoto** [reply]
難しい。。。
